src/examples/java/com/amazonaws/crypto/examples/BasicEncryptionExample.java

```diff
@@ -48,32 +52,44 @@ static void encryptAndDecrypt(final String keyArn) {
+        // 2. Instantiate a KMS Client Supplier. This example uses the default client supplier but you can
+        //    also configure the credentials provider, client configuration and other settings as necessary
+        final KmsClientSupplier clientSupplier = KmsClientSupplier.builder().build();
+
+        // 3. Instantiate a KMS Keyring, supplying the keyArn as the generator for generating a data key.
+        //    For this example, empty lists are provided for grant tokens and additional keys to encrypt the data
+        //    key with, but those can be supplied as necessary.
+        final Keyring keyring = StandardKeyrings.kms(clientSupplier, emptyList(), emptyList(), keyArn);
 
-        // 3. Create an encryption context
+        // 4. Create an encryption context
         //
-        // Most encrypted data should have an associated encryption context
-        // to protect integrity. This sample uses placeholder values.
+        //    Most encrypted data should have an associated encryption context
+        //    to protect integrity. This sample uses placeholder values.
         //
-        // For more information see:
-        // blogs.aws.amazon.com/security/post/Tx2LZ6WBJJANTNW/How-to-Protect-the-Integrity-of-Your-Encrypted-Data-by-Using-AWS-Key-Management
+        //    For more information see:
+        //    blogs.aws.amazon.com/security/post/Tx2LZ6WBJJANTNW/How-to-Protect-the-Integrity-of-Your-Encrypted-Data-by-Using-AWS-Key-Management
         final Map<String, String> encryptionContext = Collections.singletonMap("ExampleContextKey", "ExampleContextValue");
 
-        // 4. Encrypt the data
-        final CryptoResult<byte[], KmsMasterKey> encryptResult = crypto.encryptData(masterKeyProvider, EXAMPLE_DATA, encryptionContext);
+        // 5. Instantiate the AwsCryptoConfig input to AwsCrypto with the keyring and encryption context
+        final AwsCryptoConfig config = AwsCryptoConfig.builder()
+                .keyring(keyring)
+                .encryptionContext(encryptionContext)
+                .build();
+
+        // 6. Encrypt the data
+        final AwsCryptoResult<byte[]> encryptResult = crypto.encryptData(config, EXAMPLE_DATA);
         final byte[] ciphertext = encryptResult.getResult();
 
-        // 5. Decrypt the data
-        final CryptoResult<byte[], KmsMasterKey> decryptResult = crypto.decryptData(masterKeyProvider, ciphertext);
+        // 7. Decrypt the data
+        final AwsCryptoResult<byte[]> decryptResult = crypto.decryptData(config, ciphertext);
 
-        // 6. Before verifying the plaintext, verify that the customer master key that
-        // was used in the encryption operation was the one supplied to the master key provider.
-        if (!decryptResult.getMasterKeyIds().get(0).equals(keyArn)) {
+        // 8. Before verifying the plaintext, verify that the key that was used in the encryption
```

Comment:
Before returning the plaintext, verify that the same CMK was used to encrypt and decrypt the data. (This is kind of silly in symmetric encryption.)

Better: Before returning the plaintext, verify that the CMK that was used to decrypt the encrypted data key was the CMK in the encryption keyring. (Again, I can't imagine how this can fail unless both keyrings are discovery keyrings.)

Actually, the more interesting test of our keyring implementation is whether each encrypted data key was encrypted by a CMK specified in the keyring. Then, unless you specify a discovery keyring, you can verify that the CMK that decrypted the encrypted data key was one of the CMKs in the decryption keyring. 

In all of these cases, a failure means that the ESDK failed; not that the user did anything wrong.

Reply:
I've reworded this to match your suggestion. I would say that the example code isn't really meant as a test, this verification is really just to introduce the Keyring Trace so the user knows how to get the information that used to be in decryptResult.getMasterKeyIds